packages/cli/src/UserManagement/email/NodeMailer.ts

```diff
@@ -21,6 +21,12 @@ export class NodeMailer {
 			secure: config.getEnv('userManagement.emails.smtp.secure'),
 		};
 
+		if (config.getEnv('userManagement.emails.smtp.rejectUnauthorized')) {
+			transportConfig.tls = {
+				rejectUnauthorized: config.getEnv('userManagement.emails.smtp.rejectUnauthorized'),
```

Comment:
@Joffcom were you able to test whether it works?  According to the docs it says this is only valid if `requestCert` is true, but we can't seem to set it, so maybe the private key automatically adds it.

Just making sure this would solve the problem as I'm unable to test it.

![Screenshot 2024-02-21 at 15 13 49](https://github.com/n8n-io/n8n/assets/219272/b83d0dce-ad69-4085-a765-f8fb9cd27338)

Reply:
A leading enterprise customer has asked that we prioritise this fix. @Joffcom can give you more info privately if useful

Reply:
IMO we should remove all code that allows for untrusted certificates in the core application, and instead add mechanisms to allow adding trusted certificates at the OS level. 
When we set `rejectUnauthorized` to true, we are allowing **any** certificate (including a compromised one that a user might not know about), which is just as good as no-TLS.
By switching over to a process of trusting specific certificates, we make n8n less vulnerable to MITM attacks at the infrastructure level.
Here is proposed change https://github.com/n8n-io/n8n/pull/8705.

@sirdavidoff Would it possible to put me in touch with this customer? I'd like to check with them if my proposed solution would work for them instead.

Reply:
Done

Reply:
Should we close this PR then @netroy ?

Reply:
Should we close this @netroy ?

Reply:
for anyone needing custom certificate support, #8705 has been merged and released. For customers on TCP+STARTTLS connections we now have #9415. 
#9415 is now being tested by the enterprise customer. Once they conform that this works for them, we can release that as well.

Reply:
The customer has confirmed that disabling TLS like this wouldn't have addressed their issue since the connection starts as a regular unencrypted TCP connection, and then gets upgraded to TLS using the StartTLS command.

They also confirmed that #9415 seems to have fixed the issue for them. So, I'll try to get that merged and released into a patch tomorrow.